xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -1645,17 +1611,54 @@ void CVaapiRenderPicture::GLUnMapSurface()
 // Buffer Pool
 //-----------------------------------------------------------------------------
 
-VaapiBufferPool::VaapiBufferPool()
+/**
+ * Buffer pool holds allocated vaapi and gl resources
+ * Embedded in COutput
+ */
+class VAAPI::CVaapiBufferPool : public IVideoBufferPool
+{
+public:
+  CVaapiBufferPool(CDecoder &decoder);
```

Comment:
What is CDecoder here? MemoryBuffer is MemoryBuffer and should not rely on any thing like Decoder.
From theory anything could write into a buffer, not only decoders,